program/object-detection-tf-py/calc_metrics_oid.py

```diff
@@ -0,0 +1,152 @@
+#
+# Copyright (c) 2018 cTuning foundation.
+# See CK COPYRIGHT.txt for copyright details.
+#
+# SPDX-License-Identifier: BSD-3-Clause.
+# See CK LICENSE.txt for licensing details.
+#
+
+import os
+
+import numpy as np
+
+from object_detection.core import standard_fields
+from object_detection.legacy import evaluator
+from object_detection.utils import label_map_util
+
+import ck_utils
+
+def new_detection(key):
+  return {
+    'key': key,
+    'groundtruth_boxes': np.empty([0,4]),
+    'groundtruth_classes': np.array([],dtype=int),
+    'groundtruth_difficult': None,
+    'groundtruth_group_of': np.array([],dtype=int),
+    'groundtruth_image_classes': None,
+    'detection_boxes': np.empty([0,4]),
+    'detection_classes': np.array([],dtype=int),
+    'detection_scores': np.array([],dtype=float),
+  }
+
+def fill_detection(detection, file_name):
+  try:
+    with open(file_name, 'r') as f:
+      size = f.readline().split()
+      x = float(size[0])
+      y = float(size[1])
+      cnt = 0
+      for line in f:
+        det = ck_utils.Detection(line)
+        cnt += 1
+        detection['detection_boxes'] = np.append(detection['detection_boxes'], [[det.x1/x, det.y1/y, det.x2/x, det.y2/y]], 0)
+        detection['detection_classes'] = np.append(detection['detection_classes'], det.class_id)
+        detection['detection_scores'] = np.append(detection['detection_scores'], det.score)
+      # Detections dict can't be empty even there is not detection for this image
+      if cnt == 0:
+        detection['detection_boxes'] = np.array([[0.0, 0.0, 0.0, 0.0]])
+        detection['detection_classes'] = np.array([0])
+        detection['detection_scores'] = np.array([0.0])
+  except Exception:
+    print('Exception when reading data from {}'.format(detection['key'] + '.txt'))
+    detection = None
+  finally:
+    return detection
+
+def get_annotations(annotations, class_map):
+  '''
+  Get annotations from file to dictionary in memory
+
+  Args:
+    annotations: groundtruth (file with annotations)
+    class_map: dictionary for classes, text id => number id 
```

Comment:
I'd suggest to name it `class_name_to_id_map`. As Python does not have strict types, we have to name vars in a way allowing to grasp their meaning without comments.